### PR TITLE
verify GridTools against naive code generation

### PR DESCRIPTION
**Pull request by cosunae**
Technical Description
=================

framework to code generate GT and naive backend for all the integration-test/CodeGen examples, and verify one against the other one. 

Changes:
* Introduce a math function in the verifier in order to initialize the fields with some parametrized mathematical function, smooth fields, instead the so far used random data initialization that can introduce bad numerical behaviour. 
* Cmake framework to compile and code generate naive and GT for all the CodeGen examples. 
* python code generation of CodeGen is not used anymore.

### CODE REVIEW

File: src/gridtools/clang/ParamWrapper.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
```
Comment (twicki) [inline src/gridtools/clang/ParamWrapper.hpp]: upper-case filename?

File: src/gridtools/clang/arrayAddons.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#define GRIDTOOLS_CLANG_ARRAYADDONS_HPP
+
+#include <array>
+
+namespace gridtools {
+    namespace clang {
+
+        template < size_t N >
+        std::array< int, N > operator+(
+            std::array< int, N > const &a, std::array< int, N > const &b) {
```
Comment (twicki) [inline src/gridtools/clang/arrayAddons.hpp]: lhs and rhs instead of a and b?

File: src/gridtools/clang/verify.hpp [hunk 3]
```diff
@@ -46,6 +44,30 @@ namespace gridtools {
                 fill_impl(value, storages...);
             }
 
+            template < class... StorageTypes >
+            void fillMath(double a,
+                double b,
```
Comment (twicki) [inline src/gridtools/clang/verify.hpp]: did you clang-format this?

File: src/gtclang/Frontend/GTClangASTConsumer.cpp [hunk 4]
```diff
@@ -220,8 +222,7 @@ void GTClangASTConsumer::HandleTranslationUnit(clang::ASTContext& ASTContext) {
   // Remove the code from stencil-functions
   for(const auto& stencilFunPair : stencilParser.getStencilFunctionMap()) {
     clang::CXXRecordDecl* stencilFunDecl = stencilFunPair.first;
-    rewriter.ReplaceText(stencilFunDecl->getSourceRange(),
-                         "class " + stencilFunPair.second->Name + "{}");
+    rewriter.ReplaceText(stencilFunDecl->getSourceRange(), "");
```
Comment (twicki) [inline src/gtclang/Frontend/GTClangASTConsumer.cpp]: why remove this?

File: test/integration-test/CMakeLists.txt [hunk 5]
```diff
@@ -51,16 +51,54 @@ mchbuild_add_custom_dummy_target(NAME GTClangQtCreatorParseProjectIntegrationTes
```
Comment (twicki) [inline test/integration-test/CMakeLists.txt]: I need a quick rundown on how this is called

File: src/gridtools/clang/ParamWrapper.hpp (hunk 1 above)
Comment (cosunae) [reply src/gridtools/clang/ParamWrapper.hpp]: changed

File: src/gridtools/clang/verify.hpp (hunk 3 above)
Comment (twicki) [reply src/gridtools/clang/verify.hpp]: I see. Do we want to keep it that way?

File: src/gtclang/Frontend/GTClangASTConsumer.cpp (hunk 4 above)
Comment (twicki) [reply src/gtclang/Frontend/GTClangASTConsumer.cpp]: good catch

File: test/integration-test/CMakeLists.txt (hunk 5 above)
Comment (twicki) [reply test/integration-test/CMakeLists.txt]: Got it!

**Comment by twicki** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

**Pull request merged**